### ISSUE TITLE
[a11y]: TileGroup/RadioTile missing readOnly support

### ISSUE THREAD

**Issue by justindm234** [carbon-design-system/carbon#19394]
### Package

@carbon/react

### Package version

11.82.1

### Description

For proper accessibility support, input elements should support both disabled (exists) and readOnly (missing) states, since screen readers skip disabled content

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tile--selectable

### Steps to reproduce

Pass `readOnly={true}` to a RadioTile component and a `readOnly` is rendered on its LABEL element, while it should be on its INPUT element.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@justindm234 The `readOnly` attribute [doesn't apply to non-textual inputs](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/readonly#:~:text=The%20attribute%20is%20not%20relevant%20to%20all%20other%20elements%2C%20including%20%3Cselect%3E%20and%20%3Cbutton%3E.%20It%20also%20does%20not%20apply%20to%20non%2Dtextual%20input%20elements) like `<input type="radio"/>`

> screen readers skip disabled content

What screenreader are you testing with - JAWS, VoiceOver, NVDA, Talkback? IIRC voiceover reads disabled elements as "dimmed" in certain cases. Disabled elements can also be accessed with JAWS' virtual cursor.
